vaccine_feed_ingest/runners/us/giscorps_vaccine_providers/normalize.py

```diff
@@ -0,0 +1,264 @@
+    )
+
+    return schema.Access(drive=drive_bool, wheelchair=wheelchair_bool)
+
+
+def try_lookup(mapping, value, default, name=None):
+    if value is None:
+        return default
+
+    try:
+        return mapping[value]
+    except KeyError as e:
+        name = " for " + name or ""
+        logger.error("value not present in lookup table%s: %s", name, e)
+
+        return default
+
+
+def _get_published_at(site: dict) -> Optional[str]:
+    date_with_millis = site["attributes"]["CreationDate"]
+    if date_with_millis:
+        date = datetime.datetime.fromtimestamp(date_with_millis / 1000)  # Drop millis
+        return date.isoformat()
+
+    return None
+
+
+def try_get_list(lis, index, default=None):
+    if lis is None:
+        return default
+
+    try:
+        value = lis[index]
+        if value == "none":
+            logger.warn("saw none value")
+        return value
+    except IndexError:
+        return default
+
+
+def _get_normalized_location(site: dict, timestamp: str) -> schema.NormalizedLocation:
+
+    addrsplit = site["attributes"]["fulladdr"].split(", ")
+
+    zip = site["attributes"]["fulladdr"][-5:]
+    zip = zip if zip.isnumeric() else None
+
+    city_state_zip = addrsplit[1].split(" ") if try_get_list(addrsplit, 1) else None
+
+    state = site["attributes"]["State"] or None
+    state = state.strip() if state is not None else None
+
+    return schema.NormalizedLocation(
+        id=f"{SOURCE_NAME}:{_get_id(site)}",
+        name=site["attributes"]["name"],
+        address=schema.Address(
+            street1=addrsplit[0],
+            street2=None,
+            city=site["attributes"]["municipality"]
+            or try_get_list(city_state_zip, -3, default=""),
+            state=state,
+            zip=zip,
+        ),
+        location=schema.LatLng(
+            latitude=site["geometry"]["y"] if site.get("geometry") else None,
+            longitude=site["geometry"]["x"] if site.get("geometry") else None,
+        )
+        if site.get("geometry")
+        else None,  # this is convuluted. ask Adrian (the author) if theres questions
+        contact=_get_contacts(site),
+        languages=None,
+        opening_dates=None,
+        opening_hours=None,  # TODO: the format for this probably needs some mega-parsing as it looks like this -> "operhours": "Monday - Friday 8:00 am - 2:00 pm Saturdays 9:00 am - 12:00 pm",
+        availability=_get_availability(site),
+        inventory=_get_inventory(site),
+        access=_get_access(site),
+        parent_organization=None,
+        links=None,  # TODO
+        notes=_get_notes(site),
+        active=_get_active(site),
+        source=schema.Source(
+            source=SOURCE_NAME,
+            id=site["attributes"]["GlobalID"],
+            fetched_from_uri="https://services1.arcgis.com/WzFsmainVTuD5KML/ArcGIS/rest/services/COVID19_Vaccine_Site_Survey_API/FeatureServer/0",  # noqa: E501
```

Comment:
🤔 this looks like the `ak/arcgis` url.

Where was this data fetched from?

Reply:
this data was fetched from the GIScorps dataset. i didnt change this URL because all thats listed in the fetch/parse stages is the serviceItemId and i was feeling lazy and didnt want to try and get the link back out from that. I'll go find the link and swap it out